src/handler/guild/channel.ts

```diff
@@ -0,0 +1,260 @@
+import { SlashCommandBuilder } from "@discordjs/builders";
+import { ChannelType } from "discord-api-types/v10";
+import { CommandInteraction, MessageEmbed } from "discord.js";
+import { getRepository } from "typeorm";
+
+import { Command, CommandData } from "../../decorators";
+import { BotError } from "../../error";
+
+import { Guild as GuildEntity } from "../../entity/Guild";
+
+export class Channel {
+    @CommandData({ type: "guild" })
+    channelData() {
+        return new SlashCommandBuilder()
+            .setName("channel")
+            .setDescription("Configure channels")
+            .setDefaultPermission(false)
+            .addSubcommand((subcommand) =>
+                subcommand
+                    .setName("add")
+                    .setDescription("Add new channel")
+                    .addStringOption((option) =>
+                        option
+                            .setName("type")
+                            .setDescription(
+                                "Which type of channel to configure"
+                            )
+                            .setRequired(true)
+                            .addChoices([
+                                ["Log", "log"],
+                                ["Verify", "verify"],
+                                ["Troop Movements", "troop_movement"],
+                                ["Facility Updates", "land_facility"],
+                                ["Command", "command"],
+                            ])
+                    )
+                    .addChannelOption((option) =>
+                        option
+                            .setName("channel")
+                            .setDescription("Which channel to use")
```

Comment:
wouldn't it make more sense for the bot to also generate the channel instead of just configuring an existing channel to do the thing? i know the FE server already has them, but considering from the viewpoint of new channels we may add or the bot being added to other servers like yata is for torn.

Reply:
I don't know. I think it makes sense to be able to set already existing channels. Maybe there should be the option to also generate them as needed, but I'm not sure how to make that work with the slash command api

Reply:
make channel optional and if no channel is provided you generate a new one with the default name?